pkg/playback/core_playback.go

```diff
@@ -0,0 +1,210 @@
+package playback
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"gopkg.in/yaml.v2"
+)
+
+func check(err error) {
+	if err != nil {
+		panic(err)
+	}
+}
+
+// requestComparator compares 2 structs in the context of comparing a given request struct against
+// a request recorded in the request/
+// It then determines 1) whether that are equivalent 2) whether we should short-circuit
+// our search (return this one immediately, or keep scanning the cassette)
+type requestComparator func(req1 interface{}, req2 interface{}) (accept bool, shortCircuitNow bool)
+
+// A struct that knows how to serialize and de-serialize itself
+type serializable interface {
```

Comment:
This feels a bit odd to me, come to think of it -- the interface I would want for this would be the ability to just do
`thing, err := fromBytes(...)` and get `thing` (as an empty interface that I have to cast). But with this interface you have to
```go
deserializer := thing{}
thing, err := deserializer.fromBytes(...)
```
and so you're constructing this extra deserializer that isn't actually used.

What I think would be better is to change interactionReplayer so that instead of
```go
type interactionReplayer struct {
  ...
  reqType  serializable
  respType serializable
}
```

You have
```go
type interactionReplayer struct {
  ...
  serialize     func (interface{}) (bytes []byte, err error)
  deserialize func (*io.Reader) (interface{}, err error)
}
```
I *think* this is equivalent

Reply:
Totally agreed, its a clunky interface. I like the structure that you suggested. 

I ran with that and refactored the serialization in this [commit](https://github.com/stripe/stripe-cli/pull/486/commits/c7b020cc3a89d53d90ae39107b4e93a83a2de98a).